libraries/abstractions/common_io/test/test_iot_uart.c

```diff
@@ -39,38 +39,72 @@
 #define testIotUART_BAUD_RATE_BUFFER_DEFAULT_LENGTH    ( sizeof( testIotUART_BAUD_RATE_DEFAULT_STRING ) - 1 )
-#define testIotUART_WRITE_BUFFER_STRING                ( "Test for write async\n" )
-#define testIotUART_READ_BUFFER_STRING                 ( "Test for read async\n" )
-#define testIotUART_WRITE_BUFFER_LENGTH                ( sizeof( testIotUART_WRITE_BUFFER_STRING ) - 1 )
-#define testIotUART_READ_BUFFER_LENGTH                 ( sizeof( testIotUART_READ_BUFFER_STRING ) - 1 )
-#define testBufferSize                                 32
+#define testIotUART_BUFFER_STRING                      ( "Test for UART read-write\n" )
+#define testIotUART_BUFFER_LENGTH                      ( sizeof( testIotUART_BUFFER_STRING ) - 1 )
+
+#define testIotUART_BAUD_RATE_FAST             ( 115200 )
+#define testIotUART_BAUD_RATE_SLOW             ( 9600 )
+#define testIotUARTBUFFERSIZE                  ( 32 )
+#define testIotUART_BUFFER_LENGTH_LARGE        ( 200 )
+#define testIotUART_DELAY                      ( 1000 )
 /*-----------------------------------------------------------*/
 
 /*-----------------------------------------------------------*/
 /* Globals values which can be overwritten by the test
  * framework invoking these tests */
 /*-----------------------------------------------------------*/
-uint8_t ustestIotUartPort = 0;         /** Default UART port for testing */
-uint32_t ultestIotUartFlowControl = 0; /** Default UART flow control for testing */
-uint32_t ultestIotUartParity = 0;      /** Default UART parity for testing */
-uint32_t ultestIotUartWordLength = 0;  /** Default UART port for testing */
-uint32_t ultestIotUartStopBits = 0;    /** Default UART stop bits for testing */
+uint8_t ustestIotUartPort = 0; /** Default UART port for testing */
+uint32_t ultestIotUartFlowControl = 0;
+uint32_t ultestIotUartParity = 0;
+uint32_t ultestIotUartWordLength = 0;
+uint32_t ultestIotUartStopBits = 0;
 
 /*-----------------------------------------------------------*/
 /* Static Globals */
 /*-----------------------------------------------------------*/
 static SemaphoreHandle_t xtestIotUARTSemaphore = NULL;
 static StaticSemaphore_t xtestIotUARTCompleted;
+static volatile uint8_t cCancelled = 0;
 
 /**
- * @brief Application/POSIX defined callback for asynchronous read/write operation on
+ * This function sets the fast Baud rate.
+ */
+static  void    iot_uart_fast_setting(IotUARTConfig_t *config)
+{
+    config->ulBaudrate   = testIotUART_BAUD_RATE_FAST;
+    config->ulFlowControl= 0;
+    config->ulParity     = 0;
+    config->ulStopbits   = 0;
+    config->ulWordlength = 3;
+}
+
+
+/**
+ * @brief Application/POSIX defined callback for canceling an asynchronous operation.
+ * This callback function should never be called.
+ */
+static void prvCancelCallback( IotUARTOperationStatus_t xOpStatus,
+                               void * pvParams )
+{
+    BaseType_t xHigherPriorityTaskWoken;
+
+    cCancelled = 1;
```

Comment:
then remove this line

Reply:
Reverted the change, and using prvReadWriteCallback() instead. To ensure that callback is not getting called.